app/controllers/bookmarks_controller.rb

```diff
@@ -50,14 +50,10 @@ def create
       return redirect_to posts_path
     end
 
-    bookmark = Bookmark.where(reply_id: @reply.id, user_id: current_user.id, post_id: @reply.post.id,
-      type: 'reply_bookmark',).first_or_initialize
+    bookmark = Bookmark.where(reply_id: @reply.id, user_id: current_user.id, type: 'reply_bookmark').first_or_initialize
     if bookmark.new_record?
-      if params[:bookmark_name].present?
-        bookmark.update!(name: params[:bookmark_name])
-      else
-        bookmark.save!
-      end
+      params[:post_id] = @reply.post_id
+      bookmark.update!(params.permit(:name, :public, :post_id))
```

Comment:
(same as before, or even just `bookmark.post_id = @reply.post_id` before the update call: it combines unsaved changes with the ones you pass)